text/0074/0074-entity-slicing.md

```diff
@@ -0,0 +1,566 @@
+The easiest is the `SimpleEntityLoader`  api shown here. Users need only write a function for loading entities via their IDs, and Cedar handles determining which entities need to be loaded.
+
+
+```
+/// Implement this trait to load entities based on
+/// the entity manifest.
+/// This entity loader is called "Simple" for two reasons:
+/// 1) First, it may cause database consistency challenges- `load_entity` is called multiple times.
+/// 2) Second, it is not precise- the entity manifest only requires some
+/// fields to be loaded, but this loader always loads all fields.
+pub trait SimpleEntityLoader {
+    /// Simple entity loaders must implement `load_entity`,
+    /// a function that loads entities based on their `EntityUID`s.
+    /// For each element of `entity_ids`, returns the corresponding
+    /// [`Entity`] in the output vector.
+    fn load_entity(&mut self, entity_ids: impl IntoIterator<Item = &EntityUID>) -> impl Iterator<Item = Entity>;
+    
+    /// Loads all the entities needed for a request
+    /// using the `load_entity` function.
+    fn load(
+        &mut self,
+        schema: &Schema,
+        entity_manifest: &EntityManifest,
+        request: &Request,
+    ) -> Result<Entities, EntityManifestError> {
+      ...
+    }
+}
+```
+
+
+
+
+## Drawbacks
+
+### Risk in unverified code
+
+Increasing the complexity of the Cedar API with entity manifests provides new places for users to make mistakes. For example, using an outdated entity manifest can result in unsound entity slices. Another example would be a user writing a buggy implementation of entity loading using the manifest.
+
+
+### Full Cedar Language Support
+
+The current analysis does not support the full Cedar language (see the **Computing Access Paths** section). With some effort, the implementation could be extended to support all of Cedar by more carefully tracking access paths through different types of Cedar operations. However, it’s unclear just how tricky this will be.
+
+Along a similar note, all Cedar changes in the future will need a corresponding change in the static analysis, making it harder to extend Cedar. For example, any operations that apply to records will need careful consideration.
+
+
+### Supporting Partial Loading of Sets
+
+As written, entity manifests in this RFC do not support loading only parts of a Cedar Set, or only some of the ancestors in the entity hierarchy. This is because sets are loaded on the leaves of the access trie, with no way to specify which elements are requested.
+
+To support this feature, we recommend that we take a constraint-based approach. Constraints would be attached to nodes in the access trie, specifying which elements of sets or the parent hierarchy are needed. Constraints form a small query language, so this would increase the complexity of the entity manifest.
+
+
+
+
+
+# Extended examples
+
+## Paths needed in document_cloud example
+
+```
+Action::"CreateDocument"
+[
+ context.is_authenticated
+]
```

Comment:
Since we expect entity manifests to be written and consumed by machines (right?), it might make more sense to use a structured JSON format. Our experience with writing parsers for human-readable formats that support comments has shown that they are easy to mess up 😅

Reply:
I've implemented both a JSON format and a human-readable format. The human-readable format re-uses the parser for cedar expression so it's not super error prone.

Personally, I've found the human-readable format useful for understanding what's going on. For example, a human can diff a new entity manifest with an old one to understand how a new policy changes the manifest.